internal/checks/checks.go

```diff
@@ -38,15 +38,26 @@ type Error string
 
 func (e Error) Error() string { return string(e) }
 
+// TransientError is an error that can be recovered.
 type TransientError Error
 
 func (e TransientError) Error() string { return string(e) }
 
+var _ error = TransientError("")
+
+// FatalError is an error that causes the program to terminate.
+type FatalError Error
+
+func (e FatalError) Error() string { return string(e) }
+
+var _ error = FatalError("")
```

Comment:
@mem I like the changes! I'm just unsure why `var _ error` is being declared for both `FatalError` and `TransientError`?

Reply:
Blank variable declarations don't actually declare variables. They are compile-time assertions.

```
var _ interfaceType = whatever...
```

means that the compiler must validate that the result of the right-hand-side satisfies the interface `interfaceType`.

In this case, that means that `FatalError` satisfies the `error` interface (implements all its methods).

Reply:
Nice, thank you!